### ISSUE TITLE
ArgumentException when changing back a profile and deleting previous one

### ISSUE THREAD

**Issue by gilbdev** [microsoft/MixedRealityToolkit-Unity#5935]
## Describe the bug
```
ArgumentException: Object at index 0 is null
UnityEditor.SerializedObject..ctor (UnityEngine.Object[] objs, UnityEngine.Object context) (at C:/buildslave/unity/build/Editor/Mono/SerializedObject.bindings.cs:39)
UnityEditor.Editor.GetSerializedObjectInternal () (at C:/buildslave/unity/build/Editor/Mono/Inspector/Editor.cs:525)
UnityEditor.Editor.get_serializedObject () (at C:/buildslave/unity/build/Editor/Mono/Inspector/Editor.cs:432)
Microsoft.MixedReality.Toolkit.Editor.MixedRealitySceneSystemProfileInspector.OnEnable () (at Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealitySceneSystemProfileInspector.cs:67)
```

## To reproduce

~~- in mixedrealityToolkit, Section 'Scene System'
disable Scene System, have scene System Type to (None) and keep a Scene System Profile at (None)~~

~~- have the inspector reload (can happen if you rebuild any script)~~

1- from inside unity 'clone' default any profile to have a new profile
2- save

3-from git gui client or command line : discard changes on the main profile and delete the newly created profile

focus back on unity
3alternate- just reassign default profile and delete created profile from unity

4- click on Play

=> exception got spammed dozen times

## Expected behavior

no exception

## Your Setup (please complete the following information)

- Unity Version [2018.4.2f1]
- MRTK Version [v2.0.0]

## additional infos

stop and play again still displays the errors on start
closing and restarting unity fixes the problem

**Comment by gilbdev** [comment]
ok, just got the same kind of error on diagnostics. so this is not specific to scene system.

here is a 100% repro:
1- from inside unity **'clone' default *any* profile to have a new profile**
2- save 

3-from git gui client or command line : discard changes on the main profile and delete the newly created profile
- focus back on unity

3alternate- **just reassign default profile and delete created profile from unity**

4- **click on Play**
=> see 260 erros :D

- stop and play again still displays the errors on start
- closing and restarting unity fixes the problem

**Comment by Constannnnnt** [comment]
Had a similar issue.

**Setup**:
Unity Version [2019.2.6f1]
MRTK Version [v2.0.0]

**Reproduce**:
step 1: clone the hand tracking profile in the input section of the MixedRealityToolkit
step 2: disable hand mesh & joint visualization
step 3: save the profile
step 4: play. 

**Results**:
```
SerializedObjectNotCreatableException: Object at index 0 is null
UnityEditor.Editor.CreateSerializedObject () (at C:/buildslave/unity/build/Editor/Mono/Inspector/Editor.cs:577)
UnityEditor.Editor.GetSerializedObjectInternal () (at C:/buildslave/unity/build/Editor/Mono/Inspector/Editor.cs:554)
UnityEditor.Editor.get_serializedObject () (at C:/buildslave/unity/build/Editor/Mono/Inspector/Editor.cs:453)
Microsoft.MixedReality.Toolkit.Inspectors.MixedRealityHandTrackingProfileInspector.OnEnable () (at Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityHandTrackingProfileInspector.cs:28)

```
-----

I used what @gilbdev suggested: close and restart unity -> fix the issue.  From the discussion of https://forum.unity.com/threads/solved-unity-window-argumentexception-object-at-index-0-is-null.697784/ and what @wiwei said, this issue may be related to the inspector. So, right now, the best practice is restart :).

**Comment by tank-t-bird** [comment]
> I used what @gilbdev suggested: close and restart unity -> fix the issue. From the discussion of https://forum.unity.com/threads/solved-unity-window-argumentexception-object-at-index-0-is-null.697784/ and what @wiwei said, this issue may be related to the inspector. So, right now, the best practice is to restart the unity :).

Reestarting Unity worked for me. I am hitting this issue when following the [Eye Tracking Setup instructions](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/EyeTracking/EyeTracking_BasicSetup.html). 

I created a new scene and cloned the profiles as instructed and the error started showing up. Reestarting Unity fixed it as suggested in teh comment above.

I had previously made a branch with changed profiles and carried the changes over before commiting and creating the new scene, I did not have additional inspectors open.


**Setup**:
 Unity Version [2019.3.4f1]
 MRTK Version [v2.3]
